src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const stateHistory = [];
+  const newState = { ...state };
+
+  for (const keys in actions) {
```

Comment:
[CODE STYLE] - don't mutate object or arrays. You should make a copy using `Object.assign` or the spread operator before modifying it.